paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java

```diff
@@ -187,6 +187,20 @@ default boolean caseSensitive() {
         return true;
     }
 
+    /**
+     * List partitions by the given identifier
+     *
+     * <p>NOTE: cause paimon use dynamic partition all the time, Exists Partitions is not a table
+     * properties cached in memory, the method will cause io read every time it called.
+     *
+     * @param identifier path of the table
+     * @return a list of partitions of the specified table in identifier
+     * @throws TableNotExistException if the table does not exist
+     */
+    default List<Partition> listPartitions(Identifier identifier) throws TableNotExistException {
```

Comment:
Maybe we don't this method now, we can just expose `List<BinaryRow> TableScan.listPartitions`, the interface can `BinaryRow` here, because we just use `BinaryRow` in `TableWrite` too.